File: solutions/go/two-fer/1/two_fer.go
// Package twofer provides funtionality for two-for-one
package twofer

import "fmt"

// ShareWith returns One for you, one for me.
func ShareWith(name string) string {
	if len(name) == 0 {
		name = "you"
	}
	return fmt.Sprintf("One for %s, one for me.", name)
}

File: solutions/elixir/two-fer/1/lib/two_fer.ex
defmodule TwoFer do
  @doc """
  Two-fer or 2-fer is short for two for one. One for you and one for me.
  """
  @spec two_fer(String.t()) :: String.t()
  def two_fer(name) when is_binary(name) do
    "One for #{name}, one for me."
  end

  def two_fer() do
    "One for you, one for me."
  end
end
